test/quantization/test_numeric_suite_fx.py

```diff
@@ -608,47 +613,51 @@ def test_extract_weights_dynamic(self):
         }
         self._test_extract_weights(m, results_len=1, qconfig_dict=qconfig_dict)
 
-    @skipIfNoFBGEMM
-    def test_match_activations_mod(self):
+    def _test_match_activations_mod_impl(self, prepare_fn=prepare_fx):
         m = nn.Sequential(
             torch.quantization.QuantStub(),
             nn.Conv2d(1, 1, 1),
             nn.Conv2d(1, 1, 1),
         ).eval()
+        qconfig_dict = None
+        if prepare_fn == prepare_qat_fx:
+            qconfig_dict = {'': torch.quantization.get_default_qat_qconfig('fbgemm')}
         expected_occurrence = {
             ns.call_module(OutputLogger): 2,
         }
         self._test_match_activations(
             m, (torch.randn(2, 1, 2, 2),),
             prepared_expected_node_occurrence=expected_occurrence,
-            results_len=2)
+            results_len=2, qconfig_dict=qconfig_dict, prepare_fn=prepare_fn)
 
-    @override_qengines
-    def test_match_activations_fun(self):
-        class M(nn.Module):
-            def __init__(self):
-                super().__init__()
-                self.w1 = nn.Parameter(torch.empty(4, 4))
-                self.b1 = nn.Parameter(torch.zeros(4))
-                self.w2 = nn.Parameter(torch.empty(4, 4))
-                self.b2 = nn.Parameter(torch.zeros(4))
-                torch.nn.init.kaiming_uniform_(self.w1, a=math.sqrt(5))
-                torch.nn.init.kaiming_uniform_(self.w2, a=math.sqrt(5))
+    @skipIfNoFBGEMM
+    def test_match_activations_mod_ptq(self):
+        self._test_match_activations_mod_impl(prepare_fn=prepare_fx)
 
-            def forward(self, x):
-                x = F.linear(x, self.w1, self.b1)
-                x = F.linear(x, self.w2, self.b2)
-                x = F.relu(x)
-                return x
+    @skipIfNoFBGEMM
+    def test_match_activations_mod_qat(self):
+        self._test_match_activations_mod_impl(prepare_fn=prepare_qat_fx)
 
-        m = M().eval()
+    def _test_match_activations_fun_impl(self, prepare_fn=prepare_fx):
+        m = LinearReluLinearFunctional().eval()
```

Comment:
optional nit: I feel setting model to eval/train and use this to decide the rest might be more consistent with our apis